Core/src/IfcGeometryConverter/CurveConverter.h

```diff
@@ -892,10 +875,9 @@ namespace OpenInfraPlatform {
 					// Get IfcLine attributes: line point and line direction. 
 					carve::geom::vector<3> line_origin = placementConverter->convertIfcCartesianPoint(line->Pnt);
 
-					std::shared_ptr<typename IfcEntityTypesT::IfcVector> line_vec = line->Dir.lock();
+					EXPRESSReference<typename IfcEntityTypesT::IfcVector> line_vec = line->Dir;
 					if (!line_vec) {
-						BLUE_LOG(error) << line->getErrorLog() << ": No direction specified";
-						return;
+						throw oip::InconsistentGeometryException(line, " No direction specified");
 					}

```

Comment:
If `->Dir` is not an optional attribute in the schema, it cannot even happen that it is empty.

Also, please call the `convertIfcVector` function - if not there yet, introduce one in `placementConverter`.

Reply:
Resolved: 502042d